pkg/daemon/controller/controller.go

```diff
@@ -549,6 +480,45 @@ func (sc *ServiceController) waitUntilNodeReady() error {
 		})
 }
 
+// reconcileEnvironmentVariables ensures that the environment variables are set as expected on the instance.
+// Returns true if any of the environment variables were updated.
+func reconcileEnvironmentVariables(envVars map[string]string) (bool, error) {
+	// systemEnvVarRegistryPath is where system level environment variables are stored in the Windows OS
+	const systemEnvVarRegistryPath = `SYSTEM\CurrentControlSet\Control\Session Manager\Environment`
+	changeDetected := false
+	for key, expectedVal := range envVars {
+		registryKey, err := registry.OpenKey(registry.LOCAL_MACHINE, systemEnvVarRegistryPath, registry.ALL_ACCESS)
+		if err != nil {
+			return false, fmt.Errorf("unable to open Windows system registry key %s: %w", systemEnvVarRegistryPath, err)
+		}
+		defer func() { // always close the registry key, without swallowing any error returned before the defer call
+			closeErr := registryKey.Close()
+			if closeErr != nil {
+				klog.Errorf("could not close key: %v", closeErr)
+			}
+		}()
+
+		actualVal, _, err := registryKey.GetStringValue(key)
+		if err != nil && err != registry.ErrNotExist {
+			return false, fmt.Errorf("unable to read environment variable %s: %w", key, err)
+		}
+
+		if actualVal != expectedVal {
```

Comment:
This misses the corner case where the registry key doesn't exist, but it should exist with a value of `""`.
This can be covered by `err != registry.ErrNotExist || actualVal != expectedVal `

Reply:
I assume you meant `err == registry.ErrNotExist || actualVal != expectedVal`, I'll add the clause